File: HW2/HW2.py
class Solution(object):
    def firstBadVersion(self, n):
        low = 0
        high = n
        answer = -1
        while low <= high:
            mid = low + (high-low)/2
            if(isBadVersion(mid)):
                answer = mid
                high = mid-1
            if(not(isBadVersion(mid))):
                low = mid+1
        return answer

File: HW2/HW2.java
public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int low = 0;
        int high = n;
        int answer = -1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(isBadVersion(mid)){
                answer = mid; 
                high = mid-1;
            }else if(!isBadVersion(mid)){
                low = mid+1;
            }
        }
        return answer;
    }
}